x-pack/plugins/timelines/common/utils/field_formatters.ts

```diff
@@ -51,117 +57,141 @@ export const isRuleParametersFieldOrSubfield = (field: string, prependField?: st
-          },
-        ]
-      : [];
-
-    // format nested fields
-    let nestedFields: TimelineEventsDetailsItem[] = [];
-    if (isRuleParametersFieldOrSubfield(field, prependField)) {
-      nestedFields = Array.isArray(item)
-        ? item
-            .reduce<TimelineEventsDetailsItem[][]>((acc, curr) => {
-              acc.push(getDataFromFieldsHits(curr as Fields, dotField, fieldCategory));
-              return acc;
-            }, [])
-            .flat()
-        : getDataFromFieldsHits(item, dotField, fieldCategory);
-    } else {
-      nestedFields = Array.isArray(item)
-        ? item
-            .reduce<TimelineEventsDetailsItem[][]>((acc, curr) => {
-              acc.push(getDataFromFieldsHits(curr as Fields, dotField, fieldCategory));
-              return acc;
-            }, [])
-            .flat()
-        : getDataFromFieldsHits(item, prependField, fieldCategory);
+    // Handle threat enrichment
+    if (isThreatEnrichmentFieldOrSubfield(field, prependField)) {
+      const enrichmentItem = createFieldItem(fieldCategory, dotField, strArr, isObjectArray);
+      resultMap.set(dotField, enrichmentItem);
     }
 
-    // combine duplicate fields
-    const flat: Record<string, TimelineEventsDetailsItem> = [
-      ...accumulator,
-      ...nestedFields,
-      ...threatEnrichmentObject,
-    ].reduce(
-      (acc, f) => ({
-        ...acc,
-        // acc/flat is hashmap to determine if we already have the field or not without an array iteration
-        // its converted back to array in return with Object.values
-        ...(acc[f.field] != null
-          ? {
-              [f.field]: {
-                ...f,
-                originalValue: acc[f.field].originalValue.includes(f.originalValue[0])
-                  ? acc[f.field].originalValue
-                  : [...acc[f.field].originalValue, ...f.originalValue],
-                values: acc[f.field].values?.includes(f.values?.[0] || '')
-                  ? acc[f.field].values
-                  : [...(acc[f.field].values || []), ...(f.values || [])],
-              },
-            }
-          : { [f.field]: f }),
-      }),
-      {} as Record<string, TimelineEventsDetailsItem>
+    // Process nested fields
+    const nestedFields = processNestedFields(
+      item,
+      dotField,
+      fieldCategory,
+      isRuleParameters || isThreatEnrichmentFieldOrSubfield(field, prependField)
     );
+    // Merge results
+    for (const nestedItem of nestedFields) {
+      const existing = resultMap.get(nestedItem.field);
+
+      if (!existing) {
+        resultMap.set(nestedItem.field, nestedItem);
+        // eslint-disable-next-line no-continue
+        continue;
+      }
 
-    return Object.values(flat);
-  }, []);
+      // Merge values and originalValue arrays
+      const mergedValues = existing.values?.includes(nestedItem.values?.[0] || '')
+        ? existing.values
+        : [...(existing.values || []), ...(nestedItem.values || [])];
```

Comment:
This parts odd to me. Why would the nested field not have the existing fields. Would that be a problem at the ES level that we would want to raise here? It's a good safeguard, but I feel like it should never happen 🤔

Reply:
I think it's to handle the case in the test file packages/kbn-securitysolution-t-grid/src/mock/mock_event_details.ts eventHit you'll see at the bottom of the fields array multiple entries with the same 'matched.field': ['host.name'], these are what we are iterating over in this code, and ultimately both the matched.field at index and at index + 2 end up in the values/originalValue arrays